Issue running with Docker

Hey so can get this to work locally. Tried adding to my docker build so I can run it on CI.

When `flakehell lint` is called I get the following error:
```
Traceback (most recent call last):
  File "/usr/local/bin/flakehell", line 8, in <module>
    sys.exit(entrypoint())
  File "/usr/local/lib/python3.6/site-packages/flakehell/_cli.py", line 39, in entrypoint
    exit_code, msg = main(argv)
  File "/usr/local/lib/python3.6/site-packages/flakehell/_cli.py", line 31, in main
    return COMMANDS[command_name](argv=argv[1:])
  File "/usr/local/lib/python3.6/site-packages/flakehell/commands/_lint.py", line 11, in lint_command
    app.run(argv)
  File "/usr/local/lib/python3.6/site-packages/flake8/main/application.py", line 360, in run
    self._run(argv)
  File "/usr/local/lib/python3.6/site-packages/flake8/main/application.py", line 347, in _run
    self.initialize(argv)
  File "/usr/local/lib/python3.6/site-packages/flake8/main/application.py", line 328, in initialize
    self.find_plugins(config_finder)
TypeError: find_plugins() takes 1 positional argument but 2 were given
make: *** [test] Error 1
```

Here is my `pip freeze` inside the docker container:
```
arrow==0.10.0
astor==0.8.1
astroid==2.4.1
atomicwrites==1.4.0
attrs==19.3.0
aws-xray-sdk==2.4.2
backcall==0.1.0
bandit==1.6.2
boto==2.41.0
botocore==1.16.7
certifi==2020.4.5.1
chardet==3.0.4
ciso8601==2.1.3
darglint==1.3.0
ddtrace==0.37.0
decorator==4.4.2
Django==1.9.7
django-crispy-forms==1.6.0
django-dirtyfields==1.4
django-extensions==1.7.0
django-filter==0.13.0
django-ical==1.5
django-model-utils==2.5
django-queryinspect==1.1.0
django-rest-swagger==0.3.8
djangorestframework==3.3.3
djangorestframework-filters==0.8.0
docutils==0.15.2
entrypoints==0.3
eradicate==1.0
factory-boy==2.7.0
fake-factory==0.5.7
flake8==3.8.1
flake8-bandit==2.1.2
flake8-broken-line==0.2.0
flake8-bugbear==19.8.0
flake8-commas==2.0.0
flake8-comprehensions==3.2.2
flake8-debugger==3.2.1
flake8-docstrings==1.5.0
flake8-eradicate==0.3.0
flake8-isort==2.9.1
flake8-polyfill==1.0.2
flake8-quotes==2.1.2
flake8-rst-docstrings==0.0.12
flake8-string-format==0.2.3
flakehell==0.3.3
Flask==0.10.1
Flask-API==0.6.7
freezegun==0.3.15
future==0.18.2
geopy==1.11.0
gevent==1.1.2
gitdb==4.0.5
GitPython==3.1.2
greenlet==0.4.15
gunicorn==19.6.0
hammock==0.2.4
hashids==1.1.0
hiredis==0.3.1
holidays==0.9.12
icalendar==4.0.6
ics==0.4
idna==2.8
importlib-metadata==1.6.0
ipdb==0.10.3
ipython==7.14.0
ipython-genutils==0.2.0
isort==4.3.21
itsdangerous==1.1.0
jedi==0.17.0
Jinja2==2.11.2
jmespath==0.9.5
jsonpickle==1.4.1
lazy-object-proxy==1.4.3
MarkupSafe==1.1.1
mccabe==0.6.1
mock==2.0.0
more-itertools==8.2.0
msgpack==1.0.0
packaging==20.3
parso==0.7.0
pathtools==0.1.2
pbr==5.4.5
pep8-naming==0.9.1
pexpect==4.8.0
pickleshare==0.7.5
piexif==1.0.5
Pillow==3.3.0
pluggy==0.13.1
ply==3.11
prometheus-client==0.0.19
prompt-toolkit==3.0.5
psycopg2==2.7.7
ptyprocess==0.6.0
py==1.8.1
py-zipkin==0.20.0
pycodestyle==2.6.0
pydocstyle==5.0.2
pyflakes==2.2.0
Pygments==2.6.1
pyinstrument==3.1.3
pyinstrument-cext==0.2.2
pylint==2.5.2
pymongo==3.10.1
pyparsing==2.4.7
pytest==5.2.4
python-dateutil==2.8.1
pytz==2016.4
PyYAML==5.3.1
raven==6.7.0
redis==2.10.6
reload==0.4
requests==2.22.0
requests-cache==0.4.12
requests-mock==1.0.0
restructuredtext-lint==1.3.0
simplejson==3.17.0
six==1.14.0
smmap==3.0.4
snowballstemmer==2.0.0
SQLAlchemy==1.0.14
stevedore==1.32.0
termcolor==1.1.0
testfixtures==6.14.1
thriftpy==0.3.9
thriftpy2==0.4.11
toml==0.10.0
traitlets==4.3.3
typed-ast==1.4.1
typing-extensions==3.7.4.2
urllib3==1.25.9
watchdog==0.10.2
wcwidth==0.1.9
wemake-python-styleguide==0.14.0
Werkzeug==1.0.1
whitenoise==3.3.1
wrapt==1.12.1
zipp==3.1.0
```

Here's the `ls -a` of the container
```
.                  api_function.py   pyproject.toml        run_api.sh
..                 app               queues.py             run_queues.sh
.baseline          entrypoint.py     queues_entrypoint.py  session
.reloadignore      gunicorn_conf.py  queues_function.py    session.egg-info
__init__.py        init.py           rate_rules            shift_swaps
api_entrypoint.py  manage.py         run.sh
```

And finally my pyproject.toml
```
[tool.flakehell]
# optionally inherit from remote config (or local if you want)
base = "https://raw.githubusercontent.com/life4/flakehell/master/pyproject.toml"
# specify any flake8 options. For example, exclude "example.py":
exclude = ["example.py"]
# make output nice
format = "grouped"
# 80 chars aren't enough in 21 century
max_line_length = 90
# show line of source code in output
show_source = true
baseline = ".baseline"

# list of plugins and rules for them
[tool.flakehell.plugins]
wemake-python-styleguide = ["+*"]
```

Let me know if you need any more info!

--- a/flakehell/_logic/_discover.py
+++ b/flakehell/_logic/_discover.py
@@ -10,6 +10,7 @@ REX_CODE = re.compile(r'^[A-Z]{1,5}[0-9]{0,5}$')
 ALIASES = {
     'flake-mutable': ('M511', ),
     'flake8-bandit': ('S', ),
+    'flake8-django': ('DJ', ),  # they say `DJ0` prefix but codes have `DJ10`
     'flake8-mock': ('M001', ),
     'flake8-pytest': ('T003', ),
     'logging-format': ('G', ),

--- a/flakehell/_patched/_app.py
+++ b/flakehell/_patched/_app.py
@@ -4,8 +4,7 @@ from pathlib import Path
 from typing import Dict, Any, List, Optional, Tuple
 
 from flake8.main.application import Application
-from flake8.options.aggregator import aggregate_options
-from flake8.options.config import get_local_plugins
+from flake8.options.config import get_local_plugins, MergedConfigParser
 from flake8.plugins.manager import ReportFormatters
 from flake8.utils import parse_unified_diff
 
@@ -59,11 +58,24 @@ class FlakeHellApplication(Application):
         # If config isn't specified, flakehell will lookup for it
         config.__dict__.update(self.get_toml_config(config_path))
 
-        # parse CLI options and legacy flake8 configs
-        self.options, self.args = aggregate_options(
-            manager=self.option_manager,
+        # Parse CLI options and legacy flake8 configs.
+        # Based on `aggregate_options`.
+        config_parser = MergedConfigParser(
+            option_manager=self.option_manager,
             config_finder=config_finder,
-            argv=argv,
+        )
+        parsed_config = config_parser.parse()
+        config.extended_default_select = self.option_manager.extended_default_select.copy()
+        for config_name, value in parsed_config.items():
+            dest_name = config_name
+            # If the config name is somehow different from the destination name,
+            # fetch the destination name from our Option
+            if not hasattr(config, config_name):
+                dest_name = config_parser.config_options[config_name].dest
+            setattr(config, dest_name, value)
+        self.options, self.args = self.option_manager.parse_args(
+            args=argv,
+            values=config,
         )
 
         # All this goes from the original `parse_configuration_and_cli`.
@@ -76,10 +88,14 @@ class FlakeHellApplication(Application):
                 self.exit()
         self.options._running_from_vcs = False
         self.check_plugins.provide_options(
-            self.option_manager, self.options, self.args
+            optmanager=self.option_manager,
+            options=self.options,
+            extra_args=self.args,
         )
         self.formatting_plugins.provide_options(
-            self.option_manager, self.options, self.args
+            optmanager=self.option_manager,
+            options=self.options,
+            extra_args=self.args,
         )
 
     def make_file_checker_manager(self) -> None:

--- a/flakehell/_patched/_app.py
+++ b/flakehell/_patched/_app.py
@@ -6,6 +6,8 @@ from typing import Dict, Any, List, Optional, Tuple
 from flake8.main.application import Application
 from flake8.options.aggregator import aggregate_options
 from flake8.options.config import get_local_plugins
+from flake8.plugins.manager import ReportFormatters
+from flake8.utils import parse_unified_diff
 
 from ._checkers import FlakeHellCheckersManager
 from ._style_guide import FlakeHellStyleGuideManager
@@ -43,7 +45,7 @@ class FlakeHellApplication(Application):
             return Path(known.config).expanduser(), unknown
         return None, argv
 
-    def parse_configuration_and_cli(self, argv: List[str] = None) -> None:
+    def parse_configuration_and_cli(self, config_finder, argv: List[str] = None) -> None:
         # if passed `--config` with path to TOML-config, we should extract it
         # before passing into flake8 mechanisms
         config_path, argv = self.extract_toml_config_path(argv=argv)
@@ -60,11 +62,25 @@ class FlakeHellApplication(Application):
         # parse CLI options and legacy flake8 configs
         self.options, self.args = aggregate_options(
             manager=self.option_manager,
-            config_finder=self.config_finder,
-            arglist=argv,
-            values=config,
+            config_finder=config_finder,
+            argv=argv,
         )
-        super().parse_configuration_and_cli(argv=argv)
+
+        # All this goes from the original `parse_configuration_and_cli`.
+        # We can't call `super` anymore because all `Application` methods
+        # redefine everything.
+        self.running_against_diff = self.options.diff
+        if self.running_against_diff:
+            self.parsed_diff = parse_unified_diff()
+            if not self.parsed_diff:
+                self.exit()
+        self.options._running_from_vcs = False
+        self.check_plugins.provide_options(
+            self.option_manager, self.options, self.args
+        )
+        self.formatting_plugins.provide_options(
+            self.option_manager, self.options, self.args
+        )
 
     def make_file_checker_manager(self) -> None:
         self.file_checker_manager = FlakeHellCheckersManager(
@@ -74,19 +90,13 @@ class FlakeHellApplication(Application):
             checker_plugins=self.check_plugins,
         )
 
-    def find_plugins(self) -> None:
-        if self.local_plugins is None:
-            self.local_plugins = get_local_plugins(
-                self.config_finder,
-                self.prelim_opts.config,
-                self.prelim_opts.isolated,
-            )
-
-        sys.path.extend(self.local_plugins.paths)
-
-        if self.check_plugins is None:
-            self.check_plugins = FlakeHellCheckers(self.local_plugins.extension)
-        super().find_plugins()
+    def find_plugins(self, config_finder) -> None:
+        local_plugins = get_local_plugins(config_finder)
+        sys.path.extend(local_plugins.paths)
+        self.check_plugins = FlakeHellCheckers(local_plugins.extension)  # this line is changed
+        self.formatting_plugins = ReportFormatters(local_plugins.report)
+        self.check_plugins.load_plugins()
+        self.formatting_plugins.load_plugins()
 
     def make_guide(self) -> None:
         """Patched StyleGuide creation just to use FlakeHellStyleGuideManager

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -39,7 +39,7 @@ home-page="https://github.com/life4/flakehell"
 requires-python=">=3.5"
 requires=[
     "entrypoints",
-    "flake8>=3.7.5",
+    "flake8>=3.8.0",
     "pygments",
     "pylint",
     "termcolor",

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -119,7 +119,7 @@ dev = [
     "mccabe",
     "pep8-naming",
     "pylint",
-    "typing-extensions", # required by WPS but doesn't installed by DepHell
+    "typing-extensions", # required by WPS but isn't installed by DepHell
     "wemake-python-styleguide",
 
     "pytest",

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -119,6 +119,7 @@ dev = [
     "mccabe",
     "pep8-naming",
     "pylint",
+    "typing-extensions", # required by WPS but doesn't installed by DepHell
     "wemake-python-styleguide",
 
     "pytest",

